<?xml version="1.0"?>
<!-- vim: set fenc=utf-8 et sts=4 ft=xml : -->
<launch>
    <arg name="replay" default="false"/>
    <group unless="$(arg replay)">
    <node
        machine="search"
        pkg="samplereturn"
        type="camlogger.py"
        name="camlogger"
        ns="/cameras/search">
        <param name="rate" value="1.0"/>
        <param name="calib_file" value="$(find samplereturn)/calibration/D5100-search.yaml"/>
        <param name="frame_id" value="search_camera_lens"/>
        <remap from="pause_state" to="/io/pause_state"/>
        <remap from="cam_img" to="image"/>
        <remap from="cam_info" to="info"/>
        <remap from="cam_status" to="status"/>
    </node>
    </group>
    <node
        machine="search"
        pkg="sample_detection"
        type="color_name_sample_detection.py"
        name="color_name_sample_detection"
        ns="/processes/sample_detection">
        <param name="color_file" type="string" value="$(find sample_detection)/config/w2c.mat"/>
        <param name="delta" type="double" value="15.0"/>
        <param name="max_area" type="double" value="80000.0"/>
        <param name="min_area" type="double" value="100.0"/>
        <param name="max_size" type="double" value="0.4"/>
        <param name="min_size" type="double" value="0.05"/>
        <remap from="monocular_img" to="/cameras/search/image"/>
        <remap from="cam_info" to="/cameras/search/info"/>
    </node>
    <node
        machine="search"
        pkg="sample_detection"
        type="ray_to_points.py"
        name="search_ray_to_points"
        ns="/processes/sample_detection">
    </node>
    <node
        machine="search"
        pkg="detection_filter"
        type="detection_filter_node.py"
        name="search_filter"
        ns="/processes/sample_detection">
        <remap from="raw_points" to="point"/>
        <param name="filter_tolerance" type="double" value="0.50"/>
        <param name="filter_threshold" type="int" value="3"/>
    </node>

</launch>
